src/relay/context.ts

```diff
@@ -60,14 +60,31 @@ class RelayContext extends EventEmitter {
 
     this.source = this.createSource()
 
+    let sinkCreator: Promise<void>
```

Comment:
Please add documentation to this block of logic.

Reply:
done